packages/backend/src/tools/changelog/pairs.ts

```diff
@@ -0,0 +1,45 @@
+import { ContractParameters, DiscoveryOutput } from '@l2beat/discovery-types'
+
+export { getMatchingContractPairs, splitSafeContractPairs }
+
+type UnsafeContractPair = readonly [
+  ContractParameters,
+  ContractParameters | undefined,
+]
+
+type SafeContractPair = readonly [ContractParameters, ContractParameters]
+
+// What if given contract appeared between previous and current?
+// What if given contract disappeared between previous and current?
```

Comment:
We should know about it in the changelog. Both of those cases are already in our discovery DB table.

Reply:
Will be a part of changelog indexer - already implemented on draft PR